multiapps-controller-core/src/main/java/org/cloudfoundry/multiapps/controller/core/validators/parameters/RoutePartValidator.java

```diff
@@ -43,6 +27,10 @@ public String attemptToCorrect(Object routePart, final Map<String, Object> conte
         return result;
     }
 
+    protected String getRouteName(Object routePart, Map<String, Object> context) {
```

Comment:
does this need to be in a separate method? it seems to only hide what the method is actually doing with the generic 'getRouteName'

Reply:
In the frist place I placed the logic about namespace in `RoutePartValidator` class but then I realized that namespace is used only on `HostValidator` implementation, so I put it there. So If remove `getRouteName` method, I have to override the whole `attemptToCorrect` method and execute almost the same with only difference calling another method from `NameUtil` class.

Reply:
ah, I see; still, maybe make method name more descriptive i.e. modifyAndShortenRoutePart or similar;
also, just realized that
result = result.replaceAll("^(-*)", "");
result = result.replaceAll("(-*)$", "");
this code should probably happen before that method invocation, as it also has the potential of shortening the value and possibly bringing it down to acceptable size.